### DISCUSSION TITLE
Требуются волонтеры для сборки пакетов для Linux (rpm, deb, flatpak, snap и т.п.)

### DISCUSSION THREAD

**Discussion by trueromanus** [anilibria/anilibria-winmaclinux#108] [Обсуждения]
Друзья! Существует довольно большое количество различных форматов пакетов и методов их сборки. Также существует довольно много холиваров по типу snap vs deb и т.п. Проекту уже помогают несколько волонтеров которые либо собирают пакеты под конкретные дистрибутивы либо делают спецификации под них за что им ❤ОГРОМНОЕ СПАСИБО❤!!!! Я делаю пару пакетов при выпуске версий но все равно остаются не охваченными другие дистрибутивы, пакетные менеджеры и системы дистрибуции. Также пакеты которые делаю я сделаны очень плохо и класть такое в нормальную репу нормальные люди не будут. Поэтому я прошу у сообщества помощи. Если Вам нравится приложение (но это не обязательно) и Вы бы хотели сделать для него пакет для пакетного менеджера, системы дистрибуции или хз какие еще бывают варианты, то я готов со своей стороны обеспечить Вас всей необходимой информацией которая Вам потребуется для организации сборки приложения а также готов исправить (ну или хотя бы попытаться) возникающие на Вашем дистрибутиве ошибки. От Вас же потребуется желание и железная воля или знание как делаются пакеты для различных пакетных менеджеров/систем дистрибуции. Также всей работой по оформлению пакетов, ответом на комментарии (комментарии же можно оставлять под пакетом?) и прочей деятельностью в пакетном менеджере/системе дистрибуции будете заниматься Вы самостоятельно, я не буду в это вмешиваться (но и все сливки тоже будете снимать Вы 😉). За Ваше волонтерство Вы получите мое, и от потенциальных людей для кого Вы сможете поставить пакет, ❤ОГРОМНОЕ СПАСИБО❤!!!! Если у Вас есть вопросы/предложения можете писать в этой теме, также если Вы все же решите помочь со сборкой то об этом тоже стоит написать в этой теме. Заранее всем будущим волонтерам спасибо!!!!!

_Актуальные на текущий момент направления указаны в конце сообщения_

## В данный момент существуют следующие пакеты:
если Вы видите некорректные данные обязательно ~~Ударьте Меня По Носу~~отпишитесь 
### ArchLinux AUR
[anilibria-winmaclinux-git](https://aur.archlinux.org/packages/anilibria-winmaclinux-git)  - содержит всегда последние изменения из центральной ветки. Волонтер @WaffleLapkin.
[anilibria-winmaclinux](https://aur.archlinux.org/packages/anilibria-winmaclinux) - содержит стабильные сборки но не всегда последней версии.
### AltLinux
[anilibria-winmaclinux](https://packages.altlinux.org/ru/sisyphus/srpms/anilibria-winmaclinux/2882369268382192491) - содержит последнюю актуальную версию. Волонтер @Koi-foo.
### Rosa Linux
[anilibria-winmaclinux](https://abf.io/import/anilibria-winmaclinux) -  содержит последнюю актуальную версию. Волонтер @X1mper.
### Flatpak
Собранный пакет содержит самые последние на момент сборки изменения. Волонтер @KnightMurloc.

### Lunchpad пакет(ы) (там пакеты?)

**Comment by Survolog** [comment]
Привет с Rosa.
https://abf.io/import/anilibria-winmaclinux - ссылка на проект.
Устанавливать можно короткой командой
`sudo dnf in anilibria`
или более полной
`sudo dnf install anilibria-winmaclinux`
Но можно ввести в терминале anilibria сразу. Будет предложена команда установки пакета.
Или можно поискать в управлении пакетами.

Если запускаю приложение из терминала, то по закрытию у меня выдаёт ошибку сегментирования.
```
kernel: anilibria[301725]: segfault at 32050001bff0 ip 000032050001bff0 sp 00007ffd2111ade8 error 14
kernel: Code: Unable to access opcode bytes at RIP 0x32050001bfc6.
```
Не заметил, чтобы это влияло на что-либо, потому не разбирал, что это.

Что-то я зачастил обновлять этот проект последнее время в репозитории. Для Росы это не слишком типичное явление. Просто проект пока попадается на глаза. Поэтому не расстраивайтесь, если станет обновляться реже.

**Reply by trueromanus** [reply]
Спасибо за подробную информацию! Обновил заголовочное сообщение. Насчет того что будет обновляться реже не страшно, версия выходит раз в календарный месяц. Насчет ошибки при завершении приложения да она известная и проблема не в приложении к сожалению а в библиотеке Qt :(. Они уже исправили в коммерческих версиях, в опенсорс может тоже когда-нибудь заедет этот фикс.

**Comment by KnightMurloc** [comment]
могу попробовать взять flatpak если других желающих не будет.

**Reply by trueromanus** [reply]
Отлично! Если столкнетесь с проблемами в сборке пишите как ответ в этом треде, я постараюсь на все ответить и помочь чем смогу.

**Reply by Koi-foo** [reply]
А вы поставьте что-то типа 
[![Packaging status](https://repology.org/badge/vertical-allrepos/anilibria-winmaclinux.svg)](https://repology.org/project/anilibria-winmaclinux/versions)
На страничку ридми чтоб тем у кого нету в репах было стыдно :)

**Reply by trueromanus** [reply]
@Koi-foo идея интересная спасибо! Но в следующий раз пишите пожалуйста в отдельном треде потому что тут немного не про то тред. Для создания нового треда надо написать в текстовом поле в самом низу страницы.

**Reply by KnightMurloc** [reply]
ну на flathub ожидаемо не пустят. https://github.com/flathub/flathub/pull/3725#issuecomment-1358422839 в целом можно самим сборки публиковать.

**Reply by trueromanus** [reply]
Печально, но тогда вероятно нужна инструкция как делается сборка и как ее потом поставить у клиентов.

**Reply by KnightMurloc** [reply]
для сборки в нужен только tv.anilibria.app.yml там в source указан git откуда качать исходники. 
для сборки нужен flatpak-builder и org.kde.Platform (5.15-22.08) и org.kde.Sdk (5.15-22.08). 
сборка: flatpak-builder --force-clean build tv.anilibria.app.yml
установка: flatpak-builder --user --install --force-clean build tv.anilibria.app.yml 
упаковка в файл: flatpak build-bundle ~/.local/share/flatpak/repo/ anilibria.flatpak tv.anilibria.anilibria
пользователь может установить из файла: flatpak install --user anilibria.flatpak

**Reply by trueromanus** [reply]
Хорошо, попробую у себя чуть позже и оформлю в отдельную инструкцию.

**Reply by trueromanus** [reply]
Да все собирается спасибо! Отправил на проверку пользователям.

**Comment by Darkness-Fran** [comment]
Не уверен что по поводу «anilibria.flatpak» сюда писать, так что прошу прощения если ошибся.
В Telegram в опроснике «Работает версия из flatpak пакета?» нет пункта по типу «Есть маленький(е) недочёт(ы). Пришлось тыкнуть «Работает но с ошибками». А в действительности заметил лишь один моментик, хотя может быть дело в том что это Flatpak.

Не получилось настроить фон. В DEB, RPM без проблем указв путь к изображению фон сразуже меняется. В остальном проблем не заметил. Фон это не работаспособность, так что не кретично конечно. Если дело в том что это Flatpak  то ладно, изначальный светлый фон хорошо подходит. P.S. Скачивание (сохранение) торрент файлов, просмотр онлайн и синхранизация работают.

**Reply by KnightMurloc** [reply]
скорей всего проблема с изоляцией. по умолчанию приложение в flatpak не имеет доступа к основной файловой системе. нужно дать доступ к папке где лежит картинка. либо надо со стороны приложение делать открытие файла через диалог и копировать его в кэш внутри контейнера.

**Reply by trueromanus** [reply]
@KnightMurloc А разве "приложение в flatpak не имеет доступа к основной файловой системе" не исключает " приложение делать открытие файла через диалог и копировать его в кэш внутри контейнера". Если у приложения нет доступа к файловой системе то как оно скопирует файл? Или может я не понимаю чего-то.
@Darkness-Fran Раз уж Вы здесь то можно завести issue с описанием проблемы :)

**Reply by KnightMurloc** [reply]
открытие файла из flatkap происходит через портал который вызывает диалог и при выборе файла приложению даётся право на открытие этого конкретного файла. но думаю в данном случае проще всего будет просто дать приложению полный доступ к файлам.

**Reply by trueromanus** [reply]
@KnightMurloc понял, спасибо за информацию.

**Comment by Darkness-Fran** [comment]
Давно хотел попрость что бы в формате appimage вакладывался релиз. Хотя бы раз в год. Перед новым годом как вариант (так сказать подарок). Есть две причины. Во первых: ситуация с релизом 1.2.2 (ошибка синхронизации релизов). Дело в том что у меня Alt Linux 
P.S.  Большое спасибо (Сопровождающий: Evgeny Chuck) за Anilibria на Альте. Но пакет 1.2.3   всё ещё в sisyphus В p10 версия 1.2.2 (ошибка синхронизации релизов). Это конечно ситуация разовая, но всё же. Пока пользуюсь старенькой anilibria_linux_x64_1.0.1.AppImage. Во вторых: Не знаю, может ли случиться что РКН специально или случайно подрежит пути синхронизации релизов. Думаю в таком случае проблема будет решина с выходом нового релиза, а все предыдущие станут почти бесполезны, разве что через VPN пользоваться. 

Я чесно пробывал разобраться с документацией https://docs.appimage.org/ но понял что это не моё. Завёл учетку на GitHub, чтоб попрость appimage. Прочёл ваш пост «Требуются волонтеры для сборки пакетов для Linux (rpm, deb, flatpak, snap и т.п.) #108.
Теперь мне не ловко просить вас о сборке appimage, но сам я не справился.

**Reply by trueromanus** [reply]
Здравствуйте, AppImage это неплохой формат упаковки приложения. Единственный его недостаток из-за которого я перестал им собирать в свое время был в том что Gstreamer работал через пень колоду. Важный момент что AppImage это тоже изолированное окружение, и он (Gstreamer) упаковывался в пакет целиком. Gstreamer имеет модульную структуру и при использовании требует вначале загрузить свои модули и это операция по какой-то неведомой для меня причине сильно зависела от настроек в системе у пользователя, именно настроек не в изолированной части системы. У некоторых людей он просто отказывался подгружать модули из-за чего плеер не работал и в общем-то единственная причина в использовании приложения сразу отпадала. Я уже честно не помню деталей как я его собирал но в принципе Вы можете и сами попробовать собрать для этого не надо изучать талмуд с сайта по AppImage, уже есть утилита которая сокращает необходимые усилия до минимума, можете изучить ее [тут](https://github.com/probonopd/linuxdeployqt). Если что спрашивайте, помогу чем смогу. Я сам если и соберу то не в ближайшее время.

Насчет синхронизации релизов тут к сожалению защититься сложно и проблема была не в отсутствии сайтов или их не работе (к слову их можно поменять при желании через настройки), к тому же Анилибрия в следующем году переходит на новое API которое будет не совместимо с текущим что значит что старые версии отвалятся.

**Reply by Koi-foo** [reply]
> Но пакет 1.2.3 всё ещё в sisyphus В p10 версии 1.2.2 (ошибка изготовления релизов).

Пакеты проходят тестирование так как собираются для разных архитектур, это занимает определенное время.

**Reply by Darkness-Fran** [reply]
Про тестирование в sisyphus знаю. Получение самой свежей версии меня мало волнует. Просто конкретно этот случай с синхронизацией немного не удобен.

**Comment by Darkness-Fran** [comment]
Взглянул на разрешения filesystems «flatpak info --show-permissions tv.anilibria.anilibria». Заметил filesystems=xdg-config/kdeglobals:ro;
Опираясь на https://docs.flatpak.org/en/latest/sandbox-permissions-reference.html пробовал выдать разрешения на конкретную папку «/some/dir». После пробовал xdg-pictures. А он гадёныш говорит: ошибка: '/some/dir' не является командой flatpak. Вы имели в виду 'repair'? С xdg-pictures тоже самое. Помню когда то давно, уже сталкивался с этим. Как победил не помню. Ладно. Фон это рюшечки. Мелочь а приятно. Сам виноват что у меня руки кривые. :)

**Reply by trueromanus** [reply]
Пишите пожалуйста сообщения как ответ на нужный топик, для этого есть текстовое поле под последним ответом в топике. Надо создавать новый топик (текстовое поле в самом низу) для нового вопроса предложения. А то Вы на каждый ответ создаете новый топик.

**Reply by KnightMurloc** [reply]
у меня при запуске `flatpak run --filesystem=host tv.anilibria.anilibria` или `flatpak run --filesystem=xdg-pictures tv.anilibria.anilibria` работают фоны из файлов.

**Reply by KnightMurloc** [reply]
кстати что думаете. дать приложению полный доступ к файлам или только к папке с картинками или оставить как есть?

**Reply by Darkness-Fran** [reply]
По Flatpak: обе команды работают, настройки фона сохраняются как укажу. Большое спасибо за пример команд. Ну а доступ «Доступ ко всем файлам» или «Доступ к каталогу изображений XDG» чёрт знает. На мой взгляд лучше закинуть одну картинку в конкретную папку нежели давать расширенный доступ. Можно оставить уточнения в инструкции https://github.com/anilibria/anilibria-winmaclinux «Как установить и обновлять?». Мол если хотите свой фон, то есть две таких то команды, с такой то разницей.

**Reply by Darkness-Fran** [reply]
И на всякий случай в инструкции указать что для запуска из меню «пуск» потребуется изменить команду запуск. Дабы не лезть всё время в терминал. Правда дистрибутивы и DE у людей разные, так что можно просто оставить напоминание об изменениях в меню «Пуск».

**Reply by trueromanus** [reply]
@KnightMurloc я думаю стоит оставить доступ к папке с изображениями.

**Reply by KnightMurloc** [reply]
По мне так подобное допиливание напильником очень раздражает. Я бы просто дал доступ ко всем файлам по умолчанию. Ведь он всё равно есть если ставить не через flatpak.

**Reply by trueromanus** [reply]
Хорошо давай так.

**Reply by Darkness-Fran** [reply]
Как то не подумал о доступе у DEB и RPM. Я не знаю о том как делать flatpak. Если задать доступ только к одной папке проблемно, то да доступ ко всему. А если появятся ярые поклоники flatpak-изоляции то упомянуть о доступе у DEB и RPM.

**Reply by KnightMurloc** [reply]
@trueromanus окей завтра закуну pull request с фиксом. Заодно исправлю ссылку на репазиторий.

**Reply by KnightMurloc** [reply]
@Darkness-Fran все эти разрешения можно изменять как описано выше думаю кому надо сами запретят доступ.

**Comment by Darkness-Fran** [comment]
По поводу linuxdeployqt. Ему не нравится моя версия glibc. Из терминала:

linuxdeployqt  (commit deebf70), build 38 built on 2022-12-15 18:06:02 UTC
ERROR: The host system is too new.
Please run on a system with a glibc version no newer than what comes with the oldest
currently still-supported mainstream distribution (Ubuntu Bionic), which is glibc 2.27.
This is so that the resulting bundle will work on most still-supported Linux distributions.
For more information, please see
https://github.com/probonopd/linuxdeployqt/issues/340

linuxdeployqt (commit deebf70), build 38 built on 2022-12-15 18:06:02 UTC
ERROR: Хост-система слишком новая.
Пожалуйста, запустите на системе с версией glibc не новее той, что поставляется с самым старым
поддерживаемый в настоящее время основной дистрибутив (Ubuntu Bionic), а это glibc 2.27.
Это необходимо для того, чтобы полученный пакет работал на большинстве все еще поддерживаемых дистрибутивов Linux.
Для получения дополнительной информации, пожалуйста, см.
https://github.com/probonopd/linuxdeployqt/issues/340

У меня Alt Linux p10, а в нём glibc-2.32.
Кроме https://docs.appimage.org/, пробывал https://github.com/redicculus/arch2appimage. Там проблема в том что пакет (программа) должна располагаться в https://archlinux.org/packages. А Anilibria(и) живёт(ут) а AUR: https://aur.archlinux.org/packages/anilibria-winmaclinux
https://aur.archlinux.org/packages/anilibria-winmaclinux-git.

**Reply by trueromanus** [reply]
Пишите пожалуйста сообщения как ответ на нужный топик, для этого есть текстовое поле под последним ответом в топике. Надо создавать новый топик (текстовое поле в самом низу) для нового вопроса предложения. А то Вы на каждый ответ создаете новый топик.

**Reply by Darkness-Fran** [reply]
Прошу прощения. Я впервые на GitHub. Сейчас правильно ?

**Reply by trueromanus** [reply]
Сейчас да верно. По поводу топика да я вспомнил что linuxdeployqt требует сборки на старых версиях всего это была еще одна причина почему я перестал собирать AppImage.

**Reply by Darkness-Fran** [reply]
Я попробую через некоторое время в виртуалке linuxdeployqt потакать. Но не сейчас. После https://docs.appimage.org/ и  https://github.com/redicculus/arch2appimage хочется отдахнуть, а то за пару дней как то много новой информации пришлось усваивать. Хоть и не удачно.

**Reply by Darkness-Fran** [reply]
А можно ли надеяться, что после перехода на новое API.  Вы хотя бы разок сделаете appimage с ним.

**Reply by trueromanus** [reply]
У меня была настроенная виртуалка в которой собирался AppImage но она давно удалена и восстанавливать ее ради сборки раз в год вряд ли целесообразно. Возможно найдется волонтер кто возьмет это направление.

**Comment by Jaymz665** [comment]
@KnightMurloc Могу ли я спросить, не затруднит ли Вас добавить в flatpak версию поддержку arm'ок типа малинки, приставок и самртфонов? 🙏

**Reply by KnightMurloc** [reply]
Для Android отдельное приложение есть. Для приставок тоже вроде что-то было.

**Reply by Jaymz665** [reply]
у меня все это дело работает под линуксом pmos, armbian, mobian.

**Reply by KnightMurloc** [reply]
[anilibria.zip](https://github.com/anilibria/anilibria-winmaclinux/files/10417864/anilibria.zip)
собрал под aarch64 потестить толком не сумел так как нет подходящего железа а в эмуляторе слишком тормозит.

**Reply by Jaymz665** [reply]
Работает! Супер, теперь осталось дело за малым, когда подгонят интерфейс под маленькие экраны =)

**Comment by X1mper** [comment]
Скоро версия 1.2.8 будет в сизифе (Нестабильные репозитории ALT Linux)
https://packages.altlinux.org/ru/tasks/326440/
(Не знаю, куда лучше написать...)
Наверное, теперь я буду сопровождать этот пакет, если @Koi-foo не отзовется :)

**Reply by trueromanus** [reply]
Да это правильное место, спасибо за Вашу помощь!
